libs/contrib/Data/String/Parser.idr

```diff
@@ -406,3 +406,14 @@ export
 ntimes : Monad m => (n : Nat) -> ParseT m a -> ParseT m (Vect n a)
 ntimes    Z  p = pure Vect.Nil
 ntimes (S n) p = [| p :: (ntimes n p) |]
+
+
+||| Run the parser p between the opening parser `o` and the closing parser `c`,
+||| returning the result of p.
+export
+between : Monad m => ParseT m o -> ParseT m c -> ParseT m p -> ParseT m p
+between o c p = do
+  _ <- o
+  result <- p
+  _ <- c
+  pure result
```

Comment:
```suggestion
between o c p = id <$ o <*> p <* c
```

Reply:
I have a bit of a hard time understanding the suggestion - is it equivalent to 
```
between o c p = o *> p *< c
```
?

Reply:
The core idea is to never mix left & right pointed combinators. Only have `f arg1 arg2 arg3` with `<*` before unused arguments and `<*>` before used ones.